### ISSUE TITLE
'ContextEntry' object has no attribute 'sha256'

### ISSUE THREAD

**Issue by astrofage** [LLNL/Surfactant#138]
**Describe the bug**
Running surfactant it seems to throw an exception on some files. I read about this and could be something to do with some path issues maybe a file in the same dir that's named the same? I bypassed the issue by catching the exception and printing an error so I could produce a bom with the files that don't throw exceptions. 

**To Reproduce**
surfactant generate ./root.json this_will_crash.sbom.json

**Expected behavior**
sbom created with no uncaught exceptions

**Screenshots**
```
2024-02-16 14:26:22.449 | INFO     | surfactant.cmd.generate:sbom:293 - Processing /mnt/vbox/root/usr/bin
Traceback (most recent call last):
  File "/home/sprocket/proj/cytrics_venv/bin/surfactant", line 8, in <module>
    sys.exit(main())
  File "/home/sprocket/proj/cytrics_venv/lib/python3.10/site-packages/click/core.py", line 1157, in __call__
    return self.main(*args, **kwargs)
  File "/home/sprocket/proj/cytrics_venv/lib/python3.10/site-packages/click/core.py", line 1078, in main
    rv = self.invoke(ctx)
  File "/home/sprocket/proj/cytrics_venv/lib/python3.10/site-packages/click/core.py", line 1688, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/sprocket/proj/cytrics_venv/lib/python3.10/site-packages/click/core.py", line 1434, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/sprocket/proj/cytrics_venv/lib/python3.10/site-packages/click/core.py", line 783, in invoke
    return __callback(*args, **kwargs)
  File "/home/sprocket/proj/cytrics_venv/lib/python3.10/site-packages/surfactant/cmd/generate.py", line 367, in sbom
    if entry.sha256 not in file_symlinks:
AttributeError: 'ContextEntry' object has no attribute 'sha256'
```

**System Information (please complete the following information):**
(cytrics_venv) root# cat /etc/issue
Ubuntu 22.04.3 LTS \n \l

(cytrics_venv) root# surfactant version
0.0.0rc2

**Additional context**
I'm running sbom over a mounted vdi lvm2 partition

**Comment by astrofage** [comment]
this was my fix 
cytrics_venv/lib/python3.10/site-packages/surfactant/cmd/generate.py line 363

```
                            if file_is_symlink and entry.installPrefix:
                                # Track the symlink, but don't add to list of entries
                                # as it'll be processed later anyways
                                # ORIGINAL 
                                if entry.sha256 not in file_symlinks:
                                    file_symlinks[entry.sha256] = []
                                file_symlinks[entry.sha256].append(install_filepath)
                                ''' my fix (skip files and print err)
                                try:
                                    if entry.sha256 not in file_symlinks:
                                        file_symlinks[entry.sha256] = []
                                    file_symlinks[entry.sha256].append(install_filepath)
                                except Exception as ex:
                                    print(f"Exception on file {filepath}")
                                    entries.append(sw_parent)
                                    for sw in sw_children:
                                        entries.append(sw)
                                '''

                            else:
                                entries.append(sw_parent)
                                for sw in sw_children:
                                    entries.append(sw)
```

**Comment by nightlark** [comment]
Ah yes, this bug is already fixed in the `main` branch on GitHub -- https://github.com/LLNL/Surfactant/pull/131 was the PR that fixed it, along with a few other symlink related issues.

I just uploaded a new package to PyPI that has the fixes included (v0.0.0rc3 -- may need to pip install the specific version to make it update). Try it out and let me know if it still crashes on the files you're generating an SBOM for.

**Comment by astrofage** [comment]
pip install surfactant==0.0.0rc3

that seems to have worked, thanks!
